rpc/server.go

```diff
@@ -501,7 +518,14 @@ func (conn *Conn) runRequest(req boundRequest, arg reflect.Value, version int, o
 	}()
 	defer conn.srvPending.Done()
 
-	rv, err := req.Call(req.hdr.Request.Id, arg)
+	// Create a request-specific context, cancelled when the
+	// request returns.
+	//
+	// TODO(axw) provide a means for clients to cancel a request.
+	ctx, cancel := context.WithCancel(conn.context)
```

Comment:
I wonder if this might not be better inside methodCaller.Call, which would cancel the context a little sooner.

Reply:
We're dealing with a MethodCaller interface here, so it would have to be as well as, not instead of. I'll add cancellation in there too.